docs/release-notes/charmcraft-3.4.0.rst

```diff
@@ -0,0 +1,156 @@
+Charmcraft 3.4 release notes
+============================
+
+3 February 2025
+
+Learn about the new features, changes, and fixes introduced in Charmcraft 3.4.
+
+
+Requirements and compatibility
+------------------------------
+
+For development and testing, Charmcraft requires a host with a minimum of 4GB RAM
+running a Linux distribution compatible with systemd.
+
+All versions of Charmcraft require the following software:
+
+- systemd
+- `snapd <https://snapcraft.io/docs/installing-snapd>`_
+- Either `LXD`_ or `Multipass`_
+
+We recommend you install the `Charmcraft snap <https://snapcraft.io/charmcraft>`_. It
+comes comes bundled with all its dependencies.
+
+Non-snap installations of Charmcraft have the following dependencies:
+
+- Python 3.10 or higher
+- libgit2 1.7
+- `skopeo`_
+- `Spread`_
+
+
+What's new
+----------
+
+Charmcraft 3.4 brings the following features, integrations, and improvements.
+
+
+``charmcraft promote`` command
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In previous releases, switching charm revisions between channels and re-releasing them
+was a manual and time-consuming process that involved:
+
+1. Finding all charm revisions currently published on a given channel.
+2. Finding the revisions of each resource attached to that charm revision on the
+   channel.
+3. Releasing each of those charm revisions, with their appropriate resources, to a new
+   channel.
+
+Charmcraft 3.4 introduces the :ref:`ref_commands_promote` command, which promotes a
+charm on Charmhub from one channel to another. The command runs through this procedure
+on your behalf.
+
+If, for example, your charm supports three bases across two architectures, each with its
+own resource revision, the ``promote`` command would automatically release all six
+revisions to the new channel with the same resources.
+
+
+Feature deprecations
+--------------------
+
+The following features are deprecated in Charmcraft 3.4 and will be removed in
+Charmcraft 4.0.
+
+
+Windows support
+~~~~~~~~~~~~~~~
+
+Charmcraft 3.0 deprecated support for Windows, and the upcoming 4.0 major release will
```

Comment:
> 3.0

I added the `.0` because I thought it was too ambiguous to say "3", even with the understanding that major versions are always breaking.